pkg/api/v1/types.go

```diff
@@ -1837,6 +1839,20 @@ type ComponentStatusList struct {
 	Items []ComponentStatus `json:"items" description:"list of component status objects"`
 }
 
+// MetadataVolumeSource represents a volume containing metadata about a pod.
+type MetadataVolumeSource struct {
+	// Items is a list of metadata file name
+	Items []MetadataFile `json:"items",omitempty" description:"list of metadata files"`
+}
+
+// MetadataFile expresses information about a file holding pod metadata.
+type MetadataFile struct {
+	// Required: Name is the name of the file
+	Name string `json:"name,omitempty" description:"the name of the file to be created"`
+	// Required: Selects a field of the pod: only annotations, labels, name and namespace are supported.
+	FieldRef ObjectFieldSelector `json:"fieldRef" description:"selects a field of the pod to write to the volume"`
```

Comment:
Supported fields should be enumerated in the description tag using the exact syntax required (metadata.labels, etc.).